.github/workflows/cmake.yml

```diff
@@ -758,16 +759,14 @@ jobs:
     - name: Display all coverage artifacts
       run: |
         ls -R *.xml
-        echo "CODECOV_REPORTS=`ls -p *.xml | grep -v / | tr '\n' ',' | sed 's/,$//g`" >> $GITHUB_ENV
+        echo "CODECOV_REPORTS=`ls -p *.xml | grep -v / | tr '\n' ',' | sed 's/,$//g'`" >> $GITHUB_ENV
 
     - name: Upload reports
       uses: codecov/codecov-action@v4
-      if: (env.CODECOV_TOKEN != '' || github.repository == 'zlib-ng/zlib-ng')
```

Comment:
Shouldn't this still be `env.CODECOV_TOKEN != ''` for forks?

Reply:
I assume you mean to disable this job for forks that don't have CODECOV_TOKEN set?
The problem with that is that it then also hinders forks from being able to have codecov with tokenless uploads as required for pull requests from 3rd parties.

So this way forks without secrets.CODECOV_TOKEN set will get a failed upload CI job (I'd say that is a minor inconvenience at worst), but those that do care about codecov and want to be able to receive pull requests from others can have that work as well.
It could maybe be improved with tests for pushes vs pull requests though, but you'd still get a failed upload for pull requests to repos with no associated codecov account.

Only having `if: (env.CODECOV_TOKEN != ''` would break on pull requests to all repos, also ours.

Not sure there is a way to please everyone..

Reply:
Why remove this line at all?

Reply:
The problem with keeping it is that tokenless uploads for all downstream repos remains broken, and tokenless is required for pull requests. So codecov would remain broken for pull-requests to downstream repos.

Reply:
Reinstated the if as it was so that we can land this PR. That does mean that codecov remains broken for PRs to all downstream repos.